### ISSUE TITLE
formatter: JSDoc type cast loses parentheses after a prettier-ignore comment

### ISSUE THREAD

**Issue by Ploppy3** [oxc-project/oxc#26202]
## Input

```js
// prettier-ignore
const a = /** @type {string} */ (x);
const b = /** @type {string} */ (y);
const c = /** @type {number} */ (z);
```

## Config

```jsonc
{}
```

## Oxfmt output

Oxfmt version: `0.65.0`

```js
// prettier-ignore
const a = /** @type {string} */ (x);
const b = /** @type {string} */ y;
const c = /** @type {number} */ z;
```

## Prettier output

Prettier version: `3.9.6`

```js
// prettier-ignore
const a = /** @type {string} */ (x);
const b = /** @type {string} */ (y);
const c = /** @type {number} */ (z);
```

## Additional notes

**Trigger:** every JSDoc cast *after* a `// prettier-ignore` whose ignored line itself contains a cast
loses its parentheses. Removing either the `// prettier-ignore` or the cast on the ignored line makes
the output correct.

**Why this is more than a formatting diff:** without the parentheses TypeScript ignores the cast under
`checkJs`, so types change silently.

```js
/** @param {any} raw */
export function f(raw) {
  const v = /** @type {string} */ (raw);
  return v.length;
}
```

- Before formatting: `v: string`, `f` returns `number`.
- After formatting (`*/ raw`): the cast is ignored, `v: any`, `f` returns `any` — and **`tsc` reports no
  error**, so a type-check gate does not catch the regression.

Possibly related, both closed and fixed but different shapes: #20182, #20183.
